src/persist-client/src/internal/gc.rs

```diff
@@ -194,6 +195,46 @@ where
     }
 
     pub async fn gc_and_truncate(machine: &mut Machine<K, V, T, D>, req: GcReq) {
+        // There's also a bulk delete API in s3 if the performance of this
+        // becomes an issue. Maybe make Blob::delete take a list of keys?
+        //
+        // https://docs.aws.amazon.com/AmazonS3/latest/API/API_DeleteObjects.html
+        async fn delete_all(
+            blob: &(dyn Blob + Send + Sync),
+            keys: impl Iterator<Item = BlobKey>,
+            metrics: &RetryMetrics,
+            span: Span,
```

Comment:
I perhaps should have split out the code movement, but this is identical to the old `delete_all` function except for the metrics and span parameters. (The concurrent-delete logic is the same but we want it reported differently.)